{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>molecular_profile</th>\n",
       "      <th>molecular_profile_id</th>\n",
       "      <th>disease</th>\n",
       "      <th>doid</th>\n",
       "      <th>phenotypes</th>\n",
       "      <th>therapies</th>\n",
       "      <th>therapy_interaction_type</th>\n",
       "      <th>evidence_type</th>\n",
       "      <th>evidence_direction</th>\n",
       "      <th>evidence_level</th>\n",
       "      <th>...</th>\n",
       "      <th>citation</th>\n",
       "      <th>nct_ids</th>\n",
       "      <th>rating</th>\n",
       "      <th>evidence_status</th>\n",
       "      <th>evidence_id</th>\n",
       "      <th>variant_origin</th>\n",
       "      <th>last_review_date</th>\n",
       "      <th>evidence_civic_url</th>\n",
       "      <th>molecular_profile_civic_url</th>\n",
       "      <th>is_flagged</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>JAK2 V617F</td>\n",
       "      <td>64</td>\n",
       "      <td>Lymphoid Leukemia</td>\n",
       "      <td>1037.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>B</td>\n",
       "      <td>...</td>\n",
       "      <td>Levine et al., 2005, Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>accepted</td>\n",
       "      <td>1</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2023-01-09 21:46:26 UTC</td>\n",
       "      <td>https://civicdb.org/links/evidence_items/1</td>\n",
       "      <td>https://civicdb.org/links/molecular_profiles/64</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PDGFRA D842V</td>\n",
       "      <td>99</td>\n",
       "      <td>Gastrointestinal Stromal Tumor</td>\n",
       "      <td>9253.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>B</td>\n",
       "      <td>...</td>\n",
       "      <td>Lasota et al., 2004, Lab. Invest.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>accepted</td>\n",
       "      <td>2</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2023-01-09 21:46:27 UTC</td>\n",
       "      <td>https://civicdb.org/links/evidence_items/2</td>\n",
       "      <td>https://civicdb.org/links/molecular_profiles/99</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DNMT3A R882</td>\n",
       "      <td>32</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>9119.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>B</td>\n",
       "      <td>...</td>\n",
       "      <td>LaRochelle et al., 2011, Oncotarget</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>accepted</td>\n",
       "      <td>3</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2023-01-09 21:46:25 UTC</td>\n",
       "      <td>https://civicdb.org/links/evidence_items/3</td>\n",
       "      <td>https://civicdb.org/links/molecular_profiles/32</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DNMT3A R882</td>\n",
       "      <td>32</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>9119.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>B</td>\n",
       "      <td>...</td>\n",
       "      <td>Ribeiro et al., 2012, Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>accepted</td>\n",
       "      <td>4</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2023-01-09 21:46:25 UTC</td>\n",
       "      <td>https://civicdb.org/links/evidence_items/4</td>\n",
       "      <td>https://civicdb.org/links/molecular_profiles/32</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JAK2 V617F</td>\n",
       "      <td>64</td>\n",
       "      <td>Chronic Myeloid Leukemia</td>\n",
       "      <td>8552.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>B</td>\n",
       "      <td>...</td>\n",
       "      <td>Levine et al., 2005, Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>accepted</td>\n",
       "      <td>5</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2023-01-09 21:46:26 UTC</td>\n",
       "      <td>https://civicdb.org/links/evidence_items/5</td>\n",
       "      <td>https://civicdb.org/links/molecular_profiles/64</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  molecular_profile  molecular_profile_id                         disease  \\\n",
       "0        JAK2 V617F                    64               Lymphoid Leukemia   \n",
       "1      PDGFRA D842V                    99  Gastrointestinal Stromal Tumor   \n",
       "2       DNMT3A R882                    32          Acute Myeloid Leukemia   \n",
       "3       DNMT3A R882                    32          Acute Myeloid Leukemia   \n",
       "4        JAK2 V617F                    64        Chronic Myeloid Leukemia   \n",
       "\n",
       "     doid phenotypes therapies therapy_interaction_type evidence_type  \\\n",
       "0  1037.0        NaN       NaN                      NaN    Diagnostic   \n",
       "1  9253.0        NaN       NaN                      NaN    Diagnostic   \n",
       "2  9119.0        NaN       NaN                      NaN    Diagnostic   \n",
       "3  9119.0        NaN       NaN                      NaN    Diagnostic   \n",
       "4  8552.0        NaN       NaN                      NaN    Diagnostic   \n",
       "\n",
       "  evidence_direction evidence_level  ...                             citation  \\\n",
       "0           Supports              B  ...           Levine et al., 2005, Blood   \n",
       "1           Supports              B  ...    Lasota et al., 2004, Lab. Invest.   \n",
       "2           Supports              B  ...  LaRochelle et al., 2011, Oncotarget   \n",
       "3           Supports              B  ...          Ribeiro et al., 2012, Blood   \n",
       "4           Supports              B  ...           Levine et al., 2005, Blood   \n",
       "\n",
       "  nct_ids rating evidence_status  evidence_id variant_origin  \\\n",
       "0     NaN    4.0        accepted            1        Somatic   \n",
       "1     NaN    3.0        accepted            2        Somatic   \n",
       "2     NaN    2.0        accepted            3        Somatic   \n",
       "3     NaN    3.0        accepted            4        Somatic   \n",
       "4     NaN    4.0        accepted            5        Somatic   \n",
       "\n",
       "          last_review_date                          evidence_civic_url  \\\n",
       "0  2023-01-09 21:46:26 UTC  https://civicdb.org/links/evidence_items/1   \n",
       "1  2023-01-09 21:46:27 UTC  https://civicdb.org/links/evidence_items/2   \n",
       "2  2023-01-09 21:46:25 UTC  https://civicdb.org/links/evidence_items/3   \n",
       "3  2023-01-09 21:46:25 UTC  https://civicdb.org/links/evidence_items/4   \n",
       "4  2023-01-09 21:46:26 UTC  https://civicdb.org/links/evidence_items/5   \n",
       "\n",
       "                       molecular_profile_civic_url  is_flagged  \n",
       "0  https://civicdb.org/links/molecular_profiles/64       False  \n",
       "1  https://civicdb.org/links/molecular_profiles/99       False  \n",
       "2  https://civicdb.org/links/molecular_profiles/32       False  \n",
       "3  https://civicdb.org/links/molecular_profiles/32       False  \n",
       "4  https://civicdb.org/links/molecular_profiles/64       False  \n",
       "\n",
       "[5 rows x 25 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "civic_df = pd.read_csv(\"rawData/CIViC/nightly-ClinicalEvidenceSummaries.tsv\", sep='\\t')\n",
    "# print(civic_df)\n",
    "civic_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2562, 25)\n",
      "(2197, 25)\n"
     ]
    }
   ],
   "source": [
    "# civic_df['evidence_direction'].value_counts()\n",
    "\n",
    "predictive_df = civic_df.dropna(subset=['therapies'])\n",
    "print(predictive_df.shape)\n",
    "support_df = predictive_df[predictive_df['evidence_direction'] == 'Supports']\n",
    "print(support_df.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "####\n",
    "variant_df = pd.read_csv(\"rawData/CIViC/nightly-VariantSummaries.tsv\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>variant</th>\n",
       "      <th>drugs</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>reference_bases</th>\n",
       "      <th>variant_bases</th>\n",
       "      <th>representative_transcript</th>\n",
       "      <th>drug_interaction_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>T315I</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>133748283.0</td>\n",
       "      <td>133748283.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000318560.5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>E255K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>133738363.0</td>\n",
       "      <td>133738363.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ENST00000318560.5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AKT1</td>\n",
       "      <td>E17K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>14</td>\n",
       "      <td>105246551.0</td>\n",
       "      <td>105246551.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000407796.2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ALK</td>\n",
       "      <td>F1174L</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>29443695.0</td>\n",
       "      <td>29443695.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000389048.3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ALK</td>\n",
       "      <td>R1275Q</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>29432664.0</td>\n",
       "      <td>29432664.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000389048.3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>455</th>\n",
       "      <td>ALK</td>\n",
       "      <td>L1196M</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>29443631.0</td>\n",
       "      <td>29443631.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000389048.3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>456</th>\n",
       "      <td>ALK</td>\n",
       "      <td>C1156Y</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>29445258.0</td>\n",
       "      <td>29445258.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000389048.3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>457</th>\n",
       "      <td>EZH2</td>\n",
       "      <td>A692V</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>148506437.0</td>\n",
       "      <td>148506437.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ENST00000320356.2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458</th>\n",
       "      <td>EZH2</td>\n",
       "      <td>A682G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>148506467.0</td>\n",
       "      <td>148506467.0</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>ENST00000320356.2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>459</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>Y253H</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>133738357.0</td>\n",
       "      <td>133738357.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>ENST00000318560.5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>460 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene variant  drugs  clinical_significance chromosome        start  \\\n",
       "0    ABL1   T315I    NaN                    NaN          9  133748283.0   \n",
       "1    ABL1   E255K    NaN                    NaN          9  133738363.0   \n",
       "2    AKT1    E17K    NaN                    NaN         14  105246551.0   \n",
       "3     ALK  F1174L    NaN                    NaN          2   29443695.0   \n",
       "4     ALK  R1275Q    NaN                    NaN          2   29432664.0   \n",
       "..    ...     ...    ...                    ...        ...          ...   \n",
       "455   ALK  L1196M    NaN                    NaN          2   29443631.0   \n",
       "456   ALK  C1156Y    NaN                    NaN          2   29445258.0   \n",
       "457  EZH2   A692V    NaN                    NaN          7  148506437.0   \n",
       "458  EZH2   A682G    NaN                    NaN          7  148506467.0   \n",
       "459  ABL1   Y253H    NaN                    NaN          9  133738357.0   \n",
       "\n",
       "            stop reference_bases variant_bases representative_transcript  \\\n",
       "0    133748283.0               C             T         ENST00000318560.5   \n",
       "1    133738363.0               G             A         ENST00000318560.5   \n",
       "2    105246551.0               C             T         ENST00000407796.2   \n",
       "3     29443695.0               G             T         ENST00000389048.3   \n",
       "4     29432664.0               C             T         ENST00000389048.3   \n",
       "..           ...             ...           ...                       ...   \n",
       "455   29443631.0               G             T         ENST00000389048.3   \n",
       "456   29445258.0               C             T         ENST00000389048.3   \n",
       "457  148506437.0               G             A         ENST00000320356.2   \n",
       "458  148506467.0               G             C         ENST00000320356.2   \n",
       "459  133738357.0               T             C         ENST00000318560.5   \n",
       "\n",
       "     drug_interaction_type  \n",
       "0                      NaN  \n",
       "1                      NaN  \n",
       "2                      NaN  \n",
       "3                      NaN  \n",
       "4                      NaN  \n",
       "..                     ...  \n",
       "455                    NaN  \n",
       "456                    NaN  \n",
       "457                    NaN  \n",
       "458                    NaN  \n",
       "459                    NaN  \n",
       "\n",
       "[460 rows x 11 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "missense_df = variant_df.dropna(subset=['start','stop'])\n",
    "columns = ['gene', 'variant', 'drugs', 'clinical_significance', 'chromosome', 'start', 'stop', 'reference_bases', 'variant_bases', 'representative_transcript', 'drug_interaction_type']\n",
    "missense_df = pd.DataFrame(data=missense_df, columns=columns)\n",
    "missense_df.drop_duplicates(inplace=True)\n",
    "\n",
    "# check missense\n",
    "missense_df['variant'] = missense_df['variant'].str.split(' ')\n",
    "missense_df['variant'] = missense_df['variant'].apply(lambda x: x[-1])\n",
    "missense_df['variant'] = missense_df['variant'].str.split('::')\n",
    "missense_df['variant'] = missense_df['variant'].apply(lambda x: x[-1])\n",
    "\n",
    "pd.options.display.max_rows = 20\n",
    "# update: regular expression filtering\n",
    "\n",
    "missense_df = missense_df[missense_df['variant'].str.contains('[A-Z][0-9]+[A-Z]')]\n",
    "missense_df = missense_df[~missense_df['variant'].str.contains('_')]\n",
    "missense_df = missense_df[~missense_df['variant'].str.contains('DEL')]\n",
    "missense_df = missense_df[~missense_df['variant'].str.contains('INS')]\n",
    "missense_df = missense_df[~missense_df['variant'].str.contains('INST')]\n",
    "missense_df = missense_df[~missense_df['variant'].str.contains('-')]\n",
    "missense_df = missense_df[~missense_df['variant'].str.contains('/')].reset_index(drop=True)\n",
    "\n",
    "missense_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Series([], Name: drugs, dtype: int64)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "missense_df['drugs'].value_counts()\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "ML",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
